### PR TITLE
Add a 'secret' command to the agent

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

The new command show debug info about secrets. We now keep track of
where a secret handle was used to ease troubleshooting with external
secrets management service.

### Additional Notes

Command output looks like this:
```
=== Checking executable rights ===
Executable path: /home/dev/secret.py
Check Rights: OK, the executable has the correct rights

Rights Detail:
file mode: 100700
Owner username: dev-user
Group name: dev-user

=== Secrets stats ===
Number of secrets decrypted: 4
Secrets handle decrypted:
- agent_expvar: from go_expvar
- cpu_conf: from cpu
- expvar2: from go_expvar
- testtest: from datadog.yaml
```

### CODE REVIEW

File: releasenotes/notes/secret-command-linux-48263d69619c9fce.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+enhancements:
+  - |
+    Add of a "secrets" command to show information about decrypted secrets.
```
Comment (truthbk) [inline releasenotes/notes/secret-command-linux-48263d69619c9fce.yaml]: You might want to mention how we now keep track of secret references too.

**Review by truthbk — COMMENTED** [review]
Nice stuff, left a few comments but this looks great already 👍

File: pkg/secrets/info_nix.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+import (
+	"fmt"
+	"os/user"
+	"strconv"
+	"syscall"
+)
+
+func (info *SecretInfo) populateRights() {
+	err := checkRights(secretBackendCommand)
```
Comment (truthbk) [inline pkg/secrets/info_nix.go]: Should this be `err := checkRights(info.ExecutablePath)` ?  (I know they're actually the same thing in this case, but would it be semantically more correct?)

**Review by truthbk — COMMENTED** [review]
Looks good, just a small question/nit that might be safe to ignore if you wish.

File: pkg/secrets/info_nix.go (hunk 2 above)
Comment (hush-hush) [reply pkg/secrets/info_nix.go]: :+1:

**Pull request merged**